### ISSUE TITLE
Black screen after monster appearance, in "InTheDark"

### ISSUE THREAD

**Issue by BenRQ** [NPBruce/valkyrie#916]
Valkyrie version 2.1.0

After the monster DarkYoung is being generated, there is a succession of dialog box.
When the last popup is displayed, the app gets blocked, and a black screen appears.

A quick investigation shows there is a stack overflow at this point.

**Comment by BenRQ** [comment]
In this scenario (in the dark), the author used an empty event to manage the custom monster activation.

```
[EventCustomOff]
display=false
buttons=0
```

It is used only here :

```
[CustomMonsterDarkYoung]
base=MonsterDarkYoung
traits=horror beast
activation=EventCustomOff
health=30
healthperhero=10
horror=4
```

I don't understand why, but I made sure to ignore empty events in monster activations so it does not mess with the state machine of Valkyrie.

@nicolazee I'm closing this issue, but if you have some time to explain why you did that, I'm interested :)

**Comment by NicolaZee** [comment]
Thanks for doing a hot fix for this issue.
The empty event CustomEventOff looks wrong (as logically it should not be doing anything) but as I'm sure you'd agree Valkyrie can be quirky in the way it processes Mythos Events! : )
The underlying issue is the Dark Young is not spawned using a spawn event because it may appear in four places and it can move off the board and reappear on the board at a later stage.
I cannot use multiple spawns because if I do it restores its health back to the original value.
Instead of being spawned, a token is used to mark the space it appears in (with 4 tokens defined for the 4 spaces it can appear in: TokenDarkYoungA, TokenDarkYoungB, etc).
The spawning and movement is resolved by EventDarkYoungMythos which is triggered in the Mythos Stage.
This either checks for spawning or checks for moves by calling up EventDarkYoungMove.
In the move, the Dark Young monster may move off the map.
Valkyrie does not expect Custom Monsters to spawn without a corresponding spawn event and without CustomEventOff when the monster is removed from the map a phantom triggering of EventDarkYoungMove occasionally occurs.

This results in the following log entries.

quest301=\nThe monster moves 1 space East.\nIf it cannot move to an adjacent space East, it will instead move to an adjacent space South.\nNote: it can never move across a dashed yellow line.\n\nDid the monster move at least one space?\n
editor302=Notice: Setting: DarkYoungVisible to: 0
quest303=The monster vanishes into the woods. You can still hear it knocking down trees.\n\nRemove the Dark Young Monster from the Game Board
quest304=The DarkYoung moves 2 spaces toward the nearest investigator. Then it attacks the investigator in its space who has suffered the least Damage.

In short (with Valkyrie version 2.0.0) removing CustomEventOff results in a bug!
CustomEventOff is a kludge but it is (as far as I am aware) a kludge to work around an issue in Valkyrie.

One day I hope Valkyrie can deal with Custom Monsters that can appear and disappear.
One method of doing this would be to add functionality to set the current health to a stored variable.
Then the Dark Young could be removed and come back using standard activations with its health set to whatever the health had been reduced to.

**Comment by NicolaZee** [comment]
This issue of appearing and disappearing monsters is covered in the following issue.

https://github.com/NPBruce/valkyrie/issues/594

**Comment by BenRQ** [comment]
Thanks a lot for the explanation @NicolaZee makes sense to me.
#594 has some suggestions for solutions too, I will think about it for next release.
